tests/unit/rpc/RPCEngineTests.cpp

```diff
@@ -0,0 +1,513 @@
+#include "util/AsioContextTestFixture.hpp"
+#include "util/MockBackendTestFixture.hpp"
+#include "util/MockCounters.hpp"
+#include "util/MockCountersFixture.hpp"
+#include "util/MockETLServiceTestFixture.hpp"
+#include "util/MockHandlerProvider.hpp"
+#include "util/MockLoadBalancer.hpp"
+#include "util/MockPrometheus.hpp"
+#include "util/NameGenerator.hpp"
+#include "util/Taggable.hpp"
+#include "util/config/Config.hpp"
+#include "web/Context.hpp"
+#include "web/dosguard/DOSGuard.hpp"
+#include "web/dosguard/WhitelistHandler.hpp"
+
+#include <boost/json/object.hpp>
+#include <boost/json/parse.hpp>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <xrpl/protocol/ErrorCodes.h>
+
+#include <exception>
+#include <memory>
+#include <optional>
+#include <string>
+#include <variant>
+#include <vector>
+
+using namespace rpc;
+using namespace util;
+namespace json = boost::json;
+using namespace testing;
+
+namespace {
+constexpr auto FORWARD_REPLY = R"JSON({
+    "result": 
+    {
+        "status": "success",
+        "forwarded": true
+    }
+})JSON";
+}  // namespace
+
+class RPCEngineTest : public util::prometheus::WithPrometheus,
+                      public MockBackendTest,
+                      public MockCountersTest,
+                      public MockLoadBalancerTest,
+                      public SyncAsioContextTest {
+protected:
+    Config cfg = Config{json::parse(R"JSON({
+        "server": {"max_queue_size": 2},
+        "workers": 4
+    })JSON")};
+    util::TagDecoratorFactory tagFactory{cfg};
+    WorkQueue queue = WorkQueue::make_WorkQueue(cfg);
+    web::dosguard::WhitelistHandler whitelistHandler{cfg};
+    web::dosguard::DOSGuard dosGuard{cfg, whitelistHandler};
+    std::shared_ptr<MockHandlerProvider> handlerProvider = std::make_shared<MockHandlerProvider>();
+};
+
+struct RPCEngineFlowTestCaseBundle {
+    std::string testName;
+    bool isAdmin;
+    std::string method;
+    std::string params;
+    bool forwarded;
+    std::optional<bool> isTooBusy;
+    std::optional<bool> isUnknownCmd;
+    bool handlerReturnError;
+    std::optional<rpc::Status> status;
+    std::optional<boost::json::object> response;
+};
+
+struct RPCEngineFlowParameterTest : public RPCEngineTest, WithParamInterface<RPCEngineFlowTestCaseBundle> {};
+
+static auto
+generateTestValuesForParametersTest()
+{
+    auto const isAdmin = true;
+    auto const isTooBusy = true;
+    auto const neverCalledIsTooBusy = std::nullopt;
+    auto const neverCalledUnknownCmd = std::nullopt;
+    auto const forwarded = true;
+    auto const cmdUnknown = true;
+    auto const handlerReturnError = true;
+
+    return std::vector<RPCEngineFlowTestCaseBundle>{
+        {"ForwardedSubmit",
+         isAdmin,
+         "submit",
+         "{}",
+         forwarded,
+         neverCalledIsTooBusy,
+         neverCalledUnknownCmd,
+         !handlerReturnError,
+         rpc::Status{},
+         boost::json::parse(FORWARD_REPLY).as_object()},
```

Comment:
I think to improve readability it is better here to use values but with named fields initialisation:
```cpp
{.testName = "ForwardedSubmit",
  .isAdmin = true,
  .method = "submit",
...
}
```